src/re_graph/core.cljc

```diff
@@ -34,10 +34,41 @@
                                 :payload {:query query
                                           :variables variables}}]}))))
 
-(defn mutate [& args]
+#?(:clj
+   (defn ^:private sync-wrapper
+     "Wraps the given function to allow the GraphQL result to be returned
+      synchronously. Will return a GraphQL error response if the request times
+      out. Will throw if the call returns an exception."
+     [f timeout & args]
+     (let [p        (promise)
+           callback (fn [result] (deliver p result))
+           args'    (conj (vec args) callback)]
+       (apply f args')
+
+       ;; explicit timeout to avoid unreliable aborts from underlying implementations
+       (let [result (deref p timeout :timeout)]
+         (if (= :timeout result)
+           {:errors [{:status 500, :message "GraphQL request timed out", :args args}]}
```

Comment:
Is 500 an appropriate error code? I think 503 or maybe even 504 might be more suitable, if we need one at all - as this is a client error it might be misleading to assign an HTTP response status as we never actually got the response at all. I think the message could more clearly indicate that it is a client error, not a server error.

Reply:
I've removed the status code and changed the error message. Hopefully, it is more explicit now.